src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp

```diff
@@ -235,30 +238,20 @@ void DiagnosticSurfaceShader::evaluate(
             {
                 const Material::RenderData& material_data = material->get_render_data();
 
-                // Execute the OSL shader if there is one.
-                if (material_data.m_shader_group)
-                {
-                    shading_context.execute_osl_shading(
-                        *material_data.m_shader_group,
-                        shading_point);
-                }
-
                 if (material_data.m_bsdf)
                 {
-                    const Vector3f direction = -normalize(Vector3f(shading_point.get_ray().m_dir));
+                    // Compute lighting.
+                    AOVComponents components;
+                    ShadingComponents radiance;
+                    shading_context.get_lighting_engine()->compute_lighting(
```

Comment:
One possible alternative, would be to call BSDF::sample and fetch the resulting albedo from the BSDFSample.
We'd have albedo for camera rays only, but for a diagnostic mode it could be enough.